### PR TITLE
Add toggle for database deployment

### LINKED ISSUE #142: Toggle for the database resources
Currently we are deploying the database  with `helm`. We need a mechanism to condition that deployment for production, as we decided not to offer a database on production.

**Acceptance Criteria**

* The database resources can be toggled on/off using a value on the helm values.

**Implementation Notes**

* I see two options: `production: true/false` or `database.enabled: true/false`, either of those sounds good to me.
* If/else the whole file:
    * Alertmanager: https://artifacthub.io/packages/helm/prometheus-community/kube-prometheus-stack?modal=template&template=alertmanager/alertmanager.yaml
    * CaInjector: https://artifacthub.io/packages/helm/cert-manager/cert-manager?modal=template&template=cainjector-deployment.yaml
    * https://artifacthub.io/packages/helm/argo/argo-cd?modal=template&template=argocd-applicationset/metrics.yaml

### PR DESCRIPTION
**Pull request by rh-hemartin**
* Compacted all database files into one for easier toggling.
* Separated database secret as it needs to be deployed always.
* Replaced hardcoded `POSTGRES_URL` environment variable to use a variable from the values file.
* Moved defaults for database credentials to the values file.

## Notes for reviewers

* I set the default value as "true" with a TODO to change it for production, what do you think about that?

Fixes #142

### CODE REVIEW

File: charts/kubearchive/values.yaml [hunk 1]
```diff
@@ -80,8 +86,8 @@ database:
...
   postgresData:
     # database name
-    dbName: ""
+    dbName: "postgresdb"
     # database user
-    dbUser: ""
+    dbUser: "ps_user"
     # database password
-    dbPassword: ""
+    dbPassword: "P0stgr3sdbP@ssword"  # notsecret
```
Comment (mafh314) [inline charts/kubearchive/values.yaml:93]: add a comment here to mark this vale as not a secret
`dbPassword: "P0stgr3sdbP@ssword"` # notsecret

Comment (mafh314) [reply charts/kubearchive/values.yaml:93]: nvm, you already added it :)

**Review by mafh314 — APPROVED** [review]

File: charts/kubearchive/values.yaml (hunk 1 above)
Comment (rh-hemartin) [reply charts/kubearchive/values.yaml:93]: The email from RedHat security was faster :rofl:

File: charts/kubearchive/templates/api_server/api_server.yaml [hunk 2]
```diff
@@ -45,7 +45,7 @@ spec:
                   name: {{ tpl .Values.database.secret . }}
                   key: POSTGRES_PASSWORD
             - name: POSTGRES_URL
-              value: {{ tpl .Values.database.name . }}.{{ .Values.kubearchive.namespace }}.svc.cluster.local
+              value: {{ tpl .Values.database.url . }}
```
[the remark this answers is not in the document]
Comment (rh-hemartin) [reply charts/kubearchive/templates/api_server/api_server.yaml:48]: Its already covered in this PR

**Review by maruiz93 — APPROVED** [review]

**Pull request merged**